packages/ember/addon/instance-initializers/sentry-performance.ts

```diff
@@ -73,38 +86,216 @@ export function _instrumentEmberRouter(
+  now: number;
+};
+
+interface RenderEntries {
+  [name: string]: RenderEntry;
+}
+
+function processComponentRenderBefore(payload: Payload, beforeEntries: RenderEntries) {
+  const info = {
+    payload,
+    now: timestampWithMs(),
+  };
+  beforeEntries[payload.object] = info;
+}
+
+function processComponentRenderAfter(
+  payload: Payload,
+  beforeEntries: RenderEntries,
+  op: string,
+  minComponentDuration: number,
+) {
+  const begin = beforeEntries[payload.object];
+
+  if (!begin) {
+    return;
+  }
+
+  const now = timestampWithMs();
+  const componentRenderDuration = now - begin.now;
+
+  if (componentRenderDuration * 1000 >= minComponentDuration) {
+    const activeTransaction = getActiveTransaction();
+
+    activeTransaction?.startChild({
+      op,
+      description: payload.containerKey || payload.object,
+      startTimestamp: begin.now,
+      endTimestamp: now,
+    });
+  }
+}
+
+function _instrumentComponents(config: typeof environmentConfig['@sentry/ember']) {
+  const { disableInstrumentComponents, minimumComponentRenderDuration, enableComponentDefinitions } = config;
+  if (disableInstrumentComponents) {
+    return;
+  }
+
+  const minComponentDuration = minimumComponentRenderDuration ?? 2;
+
+  const beforeEntries = {} as RenderEntries;
+  const beforeComponentDefinitionEntries = {} as RenderEntries;
+
+  const subscribe = Ember.subscribe;
+  function _subscribeToRenderEvents() {
+    subscribe('render.component', {
+      before(_name: string, _timestamp: number, payload: Payload) {
+        processComponentRenderBefore(payload, beforeEntries);
+      },
+
+      after(_name: string, _timestamp: number, payload: any, _beganIndex: number) {
+        processComponentRenderAfter(payload, beforeEntries, 'ember.component.render', minComponentDuration);
+      },
+    });
+    if (enableComponentDefinitions) {
+      subscribe('render.getComponentDefinition', {
+        before(_name: string, _timestamp: number, payload: Payload) {
+          processComponentRenderBefore(payload, beforeComponentDefinitionEntries);
+        },
+
+        after(_name: string, _timestamp: number, payload: any, _beganIndex: number) {
+          processComponentRenderAfter(payload, beforeComponentDefinitionEntries, 'ember.component.definition', 0);
+        },
+      });
     }
+  }
+  _subscribeToRenderEvents();
+}
+
+function _instrumentInitialLoad(config: typeof environmentConfig['@sentry/ember']) {
+  const startName = '@sentry/ember:initial-load-start';
+  const endName = '@sentry/ember:initial-load-end';
+  if (config.disableInitialLoadInstrumentation) {
+    performance.clearMarks(startName);
+    performance.clearMarks(endName);
+    return;
+  }
+  const measureName = '@sentry/ember:initial-load';
+
+  performance.measure(measureName, startName, endName);
+  const measures = performance.getEntriesByName(measureName);
+  const measure = measures[0];
+
+  const startTimestamp = (measure.startTime + performance.timeOrigin) / 1000;
```

Comment:
Be aware that we had a lot of issues with environments where `performance` is not available. That's why in our utils we have things like `crossPlatformPerformance` function.